src/Model.jl

```diff
@@ -203,6 +203,46 @@ function Model(system::AbstractSystem; kwargs...)
     Model(parsed.lattice, parsed.atoms, parsed.positions; parsed.magnetic_moments, kwargs...)
 end
 
+"""
+    Model(model; [lattice, positions, atoms, kwargs...])
+    Model{T}(model; [lattice, positions, atoms, kwargs...])
+
+Construct an identical model to `model` with the option to change some of the contained
+parameters. If one of `lattice`, `positions`, `atoms` is missing, the respective value
+from `model` is used. This constructor is useful for changing the data type in the model
+or for changing `lattice` or `positions` in geometry/lattice optimisations.
+"""
+function Model{T}(model::Model;
+                  lattice::AbstractMatrix=model.lattice,
+                  positions::Vector{<:AbstractVector}=model.positions,
+                  atoms::Vector{<:Element}=model.atoms,
+                  kwargs...) where {T <: Real}
+    Model(T.(lattice), atoms, positions;
+          model.model_name,
+          model.n_electrons,
+          magnetic_moments=[],  # not used because symmetries explicitly given
+          terms=model.term_types,
+          model.temperature,
+          model.smearing,
+          model.εF,
+          model.spin_polarization,
+          model.symmetries,
+          # Can be safely disabled: this has been checked for model
+          disable_electrostatics_check=true,
+          kwargs...
```

Comment:
This means the kwargs can't override the other settings (eg temperature etc). Maybe do a merge here?

Reply:
Interestingly not. The `kwargs` take preference.

Reply:
Really?
```
julia> sum([]; dims=1, dims=2)
ERROR: syntax: keyword argument "dims" repeated in call to "sum" around REPL[1]:1
```

Reply:
Yeah but this does not use a `kwargs...` construct. That makes a difference.

Reply:
```
julia> kwargs = (; dims=1)
(dims = 1,)

julia> sum(; kwargs..., dims=2)
ERROR: MethodError: no method matching sum(; dims=2)
```
Woah this is wild. Is this documented anywhere?